controller/state/statedb.go

```diff
@@ -277,20 +279,42 @@ func (s *stateDB) AssignTask(ctx context.Context, req tasks.PopTask) (tasks.Task
 		return nil, fmt.Errorf("cannot assign %q status to task", req.Status.String())
 	}
 
+	var tags []interface{}
+	if req.Tags.Exists() {
+		for _, t := range strings.Split(req.Tags.Must().String(), ",") {
```

Comment:
why not make this a list of strings in IPLD directly?

Reply:
Done